### PR TITLE
Feat: add headers param to assertHttp methods

### PR DESCRIPTION
**Pull request by bobjoris**

### CODE REVIEW

File: oap-stdlib/src/main/java/oap/http/Client.java [hunk 1]
```diff
@@ -247,7 +247,8 @@ public Optional<Response> post( String uri, String content, String contentType,
         return post( uri, content, contentType, Map.of(), timeout );
     }
 
-    public Optional<Response> post( String uri, String content, String contentType, Map<String, Object> headers, long timeout ) {
+    public Optional<Response> post( String uri, String content, String contentType, Map<String, Object> headers,
+                                    long timeout ) {
```
Comment (rtimrov) [inline oap-stdlib/src/main/java/oap/http/Client.java:251]: such formatting looks not very good. Can you change it?

File: oap-stdlib/src/main/java/oap/http/Client.java [hunk 2]
```diff
@@ -554,7 +590,8 @@ public static class Response implements Closeable {
         private InputStream inputStream;
         private byte[] content = null;
```
Comment (alebastrov) [inline oap-stdlib/src/main/java/oap/http/Client.java:591]: volatile

File: oap-stdlib/src/main/java/oap/http/Client.java [hunk 3]
```diff
@@ -554,7 +590,8 @@ public static class Response implements Closeable {
         private InputStream inputStream;
```
Comment (alebastrov) [inline oap-stdlib/src/main/java/oap/http/Client.java:590]: volatile

File: oap-stdlib/src/main/java/oap/http/Client.java [hunk 4]
```diff
@@ -310,7 +311,8 @@ private Optional<Response> getResponse( HttpRequestBase request, long timeout, C
         try {
             return Optional.of( timeout == 0 ? future.get() : future.get( timeout, MILLISECONDS ) );
         } catch( ExecutionException e ) {
-            var newEx = new UncheckedIOException( request.getURI().toString(), new IOException( e.getCause().getMessage(), e.getCause() ) );
+            var newEx = new UncheckedIOException( request.getURI().toString(),
+                new IOException( e.getCause().getMessage(), e.getCause() ) );
```
[the remark this answers is not in the document]
Comment (bobjoris) [reply oap-stdlib/src/main/java/oap/http/Client.java:315]: I'll not change this part of the code :)

**Pull request merged**